tests/lti/views/oauth_test.py

```diff
@@ -213,43 +203,60 @@ def test_it_logs_an_error_if_Canvas_login_response_lacks_access_token(self,
-        # query params.
-        pyramid_request.query_string = urllib.urlencode({
-            'code': 'TEST_OAUTH_AUTHORIZATION_CODE',
-            'state': 'TEST_OAUTH_STATE',
+
+@pytest.mark.usefixtures('util')
+class TestTokenCallback(object):
+
+    """Unit tests for token_callback() only."""
+
+    def test_it_posts_an_authorization_code_request_to_canvas(self,
+                                                              pyramid_request,
+                                                              requests_fixture):
+        oauth.token_callback(pyramid_request)
+
+        requests_fixture.post.assert_called_once_with(
+            'https://TEST_CANVAS_SERVER.com/login/oauth2/token', {
+                'code': 'TEST_OAUTH_AUTHORIZATION_CODE',
+                'grant_type': 'authorization_code',
+
+                # The client secret that we retrieved from AuthData.
+                'client_secret': 'TEST_CLIENT_SECRET',
+
+                'client_id': u'TEST_OAUTH_CONSUMER_KEY',
+                'redirect_uri': 'http://TEST_LTI_SERVER.com/token_init',
+        })
+
+
+@pytest.mark.usefixtures('util')
+class TestRefreshCallback(object):
+
+    """Unit tests for refresh_callback() only."""
+
+    def test_it_posts_a_refresh_token_request_to_canvas(self,
+                                                        pyramid_request,
+                                                        requests_fixture):
+        oauth.refresh_callback(pyramid_request)
+
+        requests_fixture.post.assert_called_once_with(
+            'https://TEST_CANVAS_SERVER.com/login/oauth2/token', {
+                'refresh_token': 'TEST_OAUTH_REFRESH_TOKEN',
+                'grant_type': 'refresh_token',
+
+                # The client secret that we retrieved from AuthData.
+                'client_secret': 'TEST_CLIENT_SECRET',
+
+                'client_id': u'TEST_OAUTH_CONSUMER_KEY',
+                'redirect_uri': 'http://TEST_LTI_SERVER.com/token_init',
         })
-        return pyramid_request
-
-    @pytest.fixture
-    def util(self, patch):
-        util = patch('lti.views.oauth.util')
-
-        # When it calls the token_callback or refresh_callback routes,
-        # Canvas calls them with a course ID, user ID, etc etc packed into
-        # `state` URL query parameter. (The `state` query params is a
-        # JSON-encoded, URL-quoted dict containing course ID, user ID, etc.)
-        util.unpack_state.return_value = {
-            constants.CUSTOM_CANVAS_COURSE_ID: 'TEST_COURSE_ID',
-            constants.CUSTOM_CANVAS_USER_ID: 'TEST_USER_ID',
-            constants.OAUTH_CONSUMER_KEY: 'TEST_OAUTH_CONSUMER_KEY',
-            constants.EXT_CONTENT_RETURN_URL: 'TEST_EXT_CONTENT_RETURN_URL',
-            constants.ASSIGNMENT_TYPE: 'TEST_ASSIGNMENT_TYPE',
-            constants.ASSIGNMENT_NAME: 'TEST_ASSIGNMENT_NAME',
-            constants.ASSIGNMENT_VALUE: 'TEST_ASSIGNMENT_VALUE',
-        }
```

Comment:
The fixtures got move out of the class and to module level because they're now shared by multiple classes